src/resolvers/registries/npm-resolver.js

```diff
@@ -121,6 +132,20 @@ export default class NpmResolver extends RegistryResolver {
           validFiles.push(name);
         }
       }
+      for (const scope of Object.keys(scoped_files)) {
```

Comment:
Why do two passes instead of doing all in the same loop? You know iterate on `files` twice and then do another loop over everything inside folders. You can definitely combine the two loops over the `files` part and avoid a separate loop for files inside the folder by extracting the core logic in the loop into a separate function.